language/tools/test-generation/src/lib.rs

```diff
@@ -150,35 +148,32 @@ fn seed(seed: Option<String>) -> [u8; 32] {
     array
 }
 
-/// Run generate_bytecode for 'iterations' iterations and test each generated module
+/// Run generate_bytecode for the range passed in and test each generated module
 /// on the bytecode verifier.
-pub fn run_generation(args: Args) {
-    env_logger::init();
-    let iterations = args.num_iterations;
+/// TODO: Run infinitely if passed in workload is `None`.
+pub fn run_generation(args: &Args, workload: Option<Range<u64>>) {
+    let seed = seed(args.seed.clone());
+    let generation_options = config::module_generation_settings();
     let mut verified_programs: u64 = 0;
     let mut executed_programs: u64 = 0;
-    let mut generation_options = ModuleGeneratorOptions::default();
-    generation_options.min_table_size = 10;
-    // The more structs, and the larger the number of type parameters the more complex the
-    // functions and bytecode sequences generated. Be careful about setting these parameters too
-    // large -- this can lead to expontial increases in the size and number of struct
-    // instantiations that can be generated.
-    generation_options.max_ty_params = 3;
-    generation_options.max_functions = 5;
-    generation_options.max_structs = 5;
-    generation_options.args_for_ty_params = true;
-    generation_options.references_allowed = false;
-    // Test generation cannot currently cope with resources
-    generation_options.add_resources = false;
-    let seed = seed(args.seed);
+    let mut generated_programs: u64 = 0;
     let mut rng = StdRng::from_seed(seed);
-    for i in 0..iterations {
+
+    for i in workload.unwrap() {
```

Comment:
What is the loop iterator variable i used for?

Reply:
This loop iteration variable is used at the moment for statistics gathering, and to output the iteration if an erroring module is found.

Reply:
Just FYI, this no longer used/needed.